llvm/lib/Transforms/IPO/DynUntrustedAllocPost.cpp

```diff
@@ -212,82 +232,89 @@ class DynUntrustedAllocPost : public ModulePass {
 
           auto index = index_iter->second;
 
-          BasicBlock::iterator iter(CS.getInstruction());
-          auto prev_inst = BB->getInstList().getPrevNode(*CS.getInstruction());
+          // Set UniqueID for hook function
+          auto callInst = CS.getInstruction();
           auto id = IDG.getConstID(M);
-
           CS.setArgument(index, id);
+          
           ++total_hooks;
-          LLVM_DEBUG(errs() << "modified callsite:\n");
-          LLVM_DEBUG(errs() << *CS.getInstruction() << "\n");
 
-          if (!prev_inst)
-            continue;
+          if (remove_hooks)
+            hookList.push_back(callInst);
 
-          CallSite CSPrev(prev_inst);
-          if (!CSPrev)
-            continue;
+          // If provided a valid path, modify given instruction
+          if (!mpk_profile_path.empty()) {
+            // Get Call Instr that hook references
+            auto allocFunc = CS.getArgument(0);
+            if (isa<CallInst>(allocFunc) || isa<InvokeInst>(allocFunc)) {
+              CallBase *allocInst = cast<CallBase>(allocFunc);
 
-          LLVM_DEBUG(errs() << "Adding: "
-                            << CSPrev.getCalledFunction()->getName().data()
-                            << " callsite for allocID: " << id->getZExtValue()
-                            << "\n");
-          alloc_map.insert(std::pair<uint64_t, Instruction *>(
-              id->getZExtValue(), prev_inst));
+              // Check to see if ID is in fault set for patching
+              auto map_iter = fault_map.find(id->getZExtValue());
+              if (map_iter == fault_map.end())
+                continue;
+
+              LLVM_DEBUG(errs() << "modified callsite:\n");
+              LLVM_DEBUG(errs() << *CS.getInstruction() << "\n");
+
+              patchInstruction(M, allocInst);
+            }
+          }
         }
       }
     }
   }
 
-  void fixFaultedAllocations(Module &M, std::vector<FaultingSite> FS) {
-    if (FS.empty()) {
+  void patchInstruction(Module &M, CallBase *inst) {
+    std::string repl_name;
+    auto calledFuncName = inst->getCalledFunction()->getName().str();
+    auto repl_iter = AllocReplacementMap.find(calledFuncName);
+    if (repl_iter != AllocReplacementMap.end()) {
```

Comment:
maybe more natural to just check if the value isn't in the map and return. then repl_name can be declared and initialized below.

Reply:
also, maybe change repl_name

Reply:
Changed check and renamed variable.